### PR TITLE
Added rule and function for local_config_platform repository rule.

### PR DESCRIPTION
**Pull request by katre**
Part of #6849.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        return "@bazel_tools//platforms:ppc";
+      case ARM:
+        return "@bazel_tools//platforms:arm";
+      case AARCH64:
+        return "@bazel_tools//platforms:aarch64";
+      case S390X:
+        return "@bazel_tools//platforms:s390x";
+    }
+    // Unknown, so skip it.
+    return null;
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: Won't this crash when fed into ImmutableList? Is there a cleaner error mode?

File: src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      Path outputDirectory,
+      BlazeDirectories directories,
+      Environment env,
+      Map<String, String> markerData)
+      throws RepositoryFunctionException {
+
+    CPU hostCpu = CPU.getCurrent();
+    OS hostOs = OS.getCurrent();
+
+    List<String> constraints = generateConstraints(hostCpu, hostOs);
```
Comment (gregestren) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: This is a list to open up the possibility of more constraint types going forward?

File: src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java (hunk 1 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: There's a null check where it is used (in constraintFileContent).

It's tricky to figure out how to signal not to use anything. I could switch to using Optional, or pass in the List and optionally add an item. Thoughts?

File: src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java (hunk 2 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: Exactly.

File: src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java (hunk 1 above)
Comment (gregestren) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: Doesn't ImmutableList.of() crash if one of its inputs is null?

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java]: You're right! Fixed.

**Comment by katre** [comment]
Added tests (and had to change a lot of test setup to make that possible).

**Pull request closed without merging**